src/functions/rolePolicyFactory.ts

```diff
@@ -0,0 +1,84 @@
+import { APIGatewayAuthorizerResult, Statement } from "aws-lambda";
+import { Jwt } from "jsonwebtoken";
+import { AccessHttpVerbMap } from "../models/AccessHttpVerbMap";
+import { ILogEvent } from "../models/ILogEvent";
+import { HttpVerb } from "../services/http-verbs";
+import Role, { getLegacyRoles } from "../services/roles";
+import StatementBuilder from "../services/StatementBuilder";
+import newPolicyDocument from "./newPolicyDocument";
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+const accessToHttpVerbs: AccessHttpVerbMap = {
+  read: ["GET", "HEAD"],
+  write: ["*"],
+};
+
+const Configuration: AuthorizerConfig = {
+  CVSFullAccess: ["/*"],
+  CVSPsvTester: ["/*"],
+  CVSHgvTester: ["/*"],
+  CVSAdrTester: ["/*"],
+  CVSTirTester: ["/*"],
+  VTMAdmin: ["/*"],
+  Certs: ["/*/v1/document-retrieval", "/*/v1/document-retrieval/*"],
+  VehicleData: ["/*/v1/enquiry", "/*/v1/enquiry/*"],
+  DVLATrailers: ["/*/trailers", "/*/trailers/*"],
```

Comment:
Shouldn't these endpoints just be:
```
  Certs: ["/v1/document-retrieval", "/v1/document-retrieval/*"],
  VehicleData: ["/v1/enquiry", "/v1/enquiry/*"],
  DVLATrailers: ["/v1/trailers", "/v1/trailers/*"],
```
without the `/*` prefix with the way the APIG routes are setup currently
![image](https://user-images.githubusercontent.com/92087051/198040544-0f7e0a34-fa94-457f-9a7d-b80227a72ffa.png)

Reply:
These aren't URL paths, they're ARN paths (there's several more fragments to an ARN path including API stage and method).
The reason for the prefix `*` was to get around the string splitting already in place - it didn't correctly split the string without it.
That said there should be new tests added to confirm that - I'll add them in 👍

Reply:
@aaron-downing absolutely right - I got the number of `*`'s wrong and added one at the beginning incorrectly. Now fixed 👍